packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart

```diff
@@ -0,0 +1,248 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'ui.dart';
+
+/// Configuration options for the GoogleMaps user interface.
+@immutable
+class MapConfiguration {
+  /// Creates a new configuration instance with the given options.
+  ///
+  /// Any options that aren't passed will be null, which allows this to serve
+  /// as either a full configuration selection, or an update to an existing
+  /// configuration where only non-null values are updated.
+  const MapConfiguration({
+    this.compassEnabled,
+    this.mapToolbarEnabled,
+    this.cameraTargetBounds,
+    this.mapType,
+    this.minMaxZoomPreference,
+    this.rotateGesturesEnabled,
+    this.scrollGesturesEnabled,
+    this.tiltGesturesEnabled,
+    this.trackCameraPosition,
+    this.zoomControlsEnabled,
+    this.zoomGesturesEnabled,
+    this.liteModeEnabled,
+    this.myLocationEnabled,
+    this.myLocationButtonEnabled,
+    this.padding,
+    this.indoorViewEnabled,
+    this.trafficEnabled,
+    this.buildingsEnabled,
+  });
+
+  /// True if the compass UI should be shown.
+  final bool? compassEnabled;
+
+  /// True if the map toolbar should be shown.
+  final bool? mapToolbarEnabled;
+
+  /// The bounds to display.
+  final CameraTargetBounds? cameraTargetBounds;
```

Comment:
This is something that users might want to "remove", is there a `CameraTargetBounds.none` or similar that allows users to pass a non-null that will clear the previous value (and not preserve it?). I think we need that for the `applyDiff` method.

Reply:
There's an `unbounded` that seems like it serves that purpose; the property was non-nullable in the old class.

But now I'm realizing that all the callbacks were nullable, so that's not going to work. I'll need to make a special wrapper for those with a non-set option like we have in at least one other plugin.

Reply:
I think I was looking at the wrong file (maybe on my phone?) when I left that last comment. Looking again, there are no callbacks in this object, and checking again everything in the old `_GoogleMapOptions` was non-null. So nothing needs a special wrapper, and it should be correct as is.

So this should be ready for a re-review.